File: golang/mygcd.go
package main

import (
	"fmt"
)

// gcd calculates the greatest common divisor (GCD) using the Euclidean algorithm.
func gcd(a, b int) int {
	for b != 0 {
		a, b = b, a%b // Simultaneous assignment to update a and b
	}
	return a
}

func main() {
	// Example usage
	num1 := 48
	num2 := 18
	result1 := gcd(num1, num2)
	fmt.Printf("The GCD of %d and %d is: %d\n", num1, num2, result1) // Output: The GCD of 48 and 18 is: 6

	num3 := 20
	num4 := 8
	result2 := gcd(num3, num4)
	fmt.Printf("The GCD of %d and %d is: %d\n", num3, num4, result2) // Output: The GCD of 20 and 8 is: 4
}

File: python/mygcd.py
def gcd(a, b):
    """
    Calculates the Greatest Common Divisor (GCD) of two numbers 
    using the iterative Euclidean algorithm.
    """
    while b != 0:
        # Update a and b: a becomes b, and b becomes the remainder of a % b
        a, b = b, a % b
    return a

if __name__ == "__main__":
    # Main section for testing the function
    num1 = 54
    num2 = 24
    result = gcd(num1, num2)
    print(f"The GCD of {num1} and {num2} is: {result}") # Output: The GCD of 54 and 24 is: 6

    num3 = 101
    num4 = 49
    result2 = gcd(num3, num4)
    print(f"The GCD of {num3} and {num4} is: {result2}") # Output: The GCD of 101 and 49 is: 1
